### ISSUE TITLE
Accessibility: StartUpModal `dialog` and `button` issues

### ISSUE THREAD

**Issue by jason-me** [stakwork/sphinx-tribes-frontend#1235]
# Accessibility: StartUpModal does not follow the established design pattern for an accessible dialog

## Issue
When a user selects the "Get Sphinx" `button` the "StartUpModal" is displayed. The modal is missing an accessible name and role, steps are styled as headings but not marked up as semantic headings, and the close "x" button does not follow the established design pattern for buttons. 

## Expected Behavior
- [WCAG 2.2 A 4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG22/#name-role-value) requires that programmatically determined roles, states, and properties are communicated to screen reader users.
- [WAI - ARIA APG Dialog(Modal) Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/).
- [WAI - ARIA APG Button Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/button/)

### For `dialog`s ###
- Screen reader users expect to hear the correct dialog role announced along with the name of the dialog so they understand its purpose.
- Keyboard users expect to encounter the [expected keyboard interaction for dialogs](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#keyboardinteraction)

### For `button`s ###
- Screen reader users expect to hear the `button` role announced along with its accessible name (e.g. "Close"), not the filename of the icon.
- Keyboard users expect to encounter the [expected keyboard interaction for buttons](https://www.w3.org/WAI/ARIA/apg/patterns/button/#keyboardinteraction) 

### For `dialog`s ###
- [ ] The element that serves as the dialog container has `role="dialog"`.
- [ ] All elements required to operate the dialog are descendants of the element that has role `dialog`.
- [ ] The dialog container element has `aria-modal="true"`.
- [ ] The dialog has an accessible name specified by either `aria-label` or in the case of image buttons `alt` (e.g. `aria-label="close"` or `alt="close"`).
- [ ] Test with a screen reader JAWS, VoiceOver, NVDA, Orca, Talkback, etc. - ensure the correct name, `dialog` role, and properties are announced.
- [ ] Test with a keyboard to ensure the pattern follows the [expected keyboard interaction for dialogs](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#keyboardinteraction))
-  [ ] Verify in Chrome and Firefox with and without a mobile viewport enabled to ensure visual look and feel is unchanged.

### For `button`s ###
- [ ] The button is a native `button` element or has `role="button"`.
- [ ] The button has an accessible label. By default, the accessible name is computed from any text content inside the button element. However, it can also be provided with `aria-labelledby` or `aria-label`.
- [ ] If a description of the button's function is present, the button element has `aria-describedby` set to the `id` of the element containing the description.
- [ ] When the action associated with a button is unavailable, the button has `aria-disabled` set to `true`.
- [ ] Test with a screen reader JAWS, VoiceOver, NVDA, Orca, Talkback, etc. - ensure the correct name and `button` role are announced.
- [ ] Test with a keyboard to ensure the pattern follows the [expected keyboard interaction for buttons](https://www.w3.org/WAI/ARIA/apg/patterns/button/#keyboardinteraction) 
-  [ ] Verify in Chrome and Firefox with and without a mobile viewport enabled to ensure visual look and feel is unchanged.

_Linked to tracking issue: #1233_
_[Bounty#3989](https://community.sphinx.chat/bounty/3989)_

**Comment by jason-me** [comment]
Closed by: #1381
